test/message/test_plan.py

```diff
@@ -226,30 +381,47 @@ def plan_abs(data):
     x_abs = p.get()
     assert (x_abs == th.tensor([1, 9, 3])).all()
 
-    me.is_client_worker = True
-
-
-def test_fetch_plan(hook):
-    hook.local_worker.is_client_worker = False
-    device_4 = sy.VirtualWorker(hook, id="device_4")
-
-    @sy.func2plan(args_shape=[(1,)])
-    def plan_mult_3(data):
-        return data * 3
 
-    sent_plan = plan_mult_3.send(device_4)
+def test_stateful_plan_multiple_workers(workers):
+    me, bob, alice = workers["me"], workers["bob"], workers["alice"]
 
-    # Fetch plan
-    fetched_plan = device_4.fetch_plan(sent_plan.id)
-    get_plan = sent_plan.get()
+    @sy.func2plan(args_shape=[(1,)], state={"bias": th.tensor([1])})
+    def plan_abs(x, state):
+        bias = state.read("bias")
+        x = x.abs()
+        return x + bias
 
-    # Execut it locally
-    x = th.tensor([-1, 2, 3])
-    assert (get_plan(x) == th.tensor([-3, 6, 9])).all()
-    assert fetched_plan.is_built
-    assert (fetched_plan(x) == th.tensor([-3, 6, 9])).all()
+    plan_abs.send(bob, alice)
+    x_ptr = th.tensor([-1, 7, 3]).send(bob)
+    p = plan_abs(x_ptr)
+    x_abs = p.get()
+    assert (x_abs == th.tensor([2, 8, 4])).all()
 
-    hook.local_worker.is_client_worker = True
+    x_ptr = th.tensor([-1, 9, 3]).send(alice)
+    p = plan_abs(x_ptr)
+    x_abs = p.get()
+    assert (x_abs == th.tensor([2, 10, 4])).all()
+
+
+# TODO: Clarify this test
+# def test_fetch_plan(hook):
```

Comment:
The idea of fetching a plan is to get a copy of an existing plan that I (the worker) can execute. We need this for providing secure serving for Grid (which I'm working right now), I'm happy to discuss this further on slack.